tools/linker/MonoTouch.Tuner/PreserveSmartEnumConversions.cs

```diff
@@ -142,20 +174,48 @@ void ProcessAttributeProvider (ICustomAttributeProvider provider, MethodDefiniti
 				if (cache == null)
 					cache = new Dictionary<TypeDefinition, Tuple<MethodDefinition, MethodDefinition>> ();
 				cache.Add (managedEnumType, pair);
+#if NET
+				Mark (pair);
+#else
 				Preserve (pair, conditionA, conditionB);
+#endif
 			}
 		}
 
 		protected override void Process (MethodDefinition method)
 		{
+#if NET
+			static bool IsPropertyMethod (MethodDefinition method)
+			{
+				return (method.SemanticsAttributes & MethodSemanticsAttributes.Getter) != 0 ||
+					(method.SemanticsAttributes & MethodSemanticsAttributes.Setter) != 0;				
```

Comment:
simpler (more readable) to call `IsGetter` or `IsSetter` on the `method`

https://github.com/jbevain/cecil/blob/634a37d3f2be7b8c72a612fce7b4907a12533d4c/Mono.Cecil/MethodDefinition.cs#L452